pkg/gui/files_panel.go

```diff
@@ -1,418 +1,910 @@
+
+		gui.Log.Errorf("Failed to getSelectedFile at handleFileRemove: %s\n", err)
 		return err
 	}
-	var deleteVerb string
+
 	if file.Tracked {
 		deleteVerb = gui.Tr.SLocalize("checkout")
 	} else {
 		deleteVerb = gui.Tr.SLocalize("delete")
 	}
+
 	message := gui.Tr.TemplateLocalize(
 		"SureTo",
 		Teml{
 			"deleteVerb": deleteVerb,
 			"fileName":   file.Name,
 		},
 	)
-	return gui.createConfirmationPanel(g, v, strings.Title(deleteVerb)+" file", message, func(g *gocui.Gui, v *gocui.View) error {
-		if err := gui.GitCommand.RemoveFile(file); err != nil {
-			return err
-		}
-		return gui.refreshFiles(g)
-	}, nil)
+
+	err = gui.createConfirmationPanel(v, strings.Title(deleteVerb)+" file", message,
+		func(g *gocui.Gui, v *gocui.View) error {
+
+			err := gui.GitCommand.RemoveFile(file)
+			if err != nil {
+				gui.Log.Errorf("Failed to remove file: %s\n", err)
+				return err
+			}
+
+			err = gui.refreshFiles()
+			if err != nil {
+				gui.Log.Errorf("Failed to refresh files: %s\n", err)
+				return err
+			}
+
+			return nil
+		}, nil)
+	if err != nil {
+		gui.Log.Errorf("Failed to createConfirmationPanel at handleFileRemove: %s\n", err)
+		return err
+	}
+
+	return nil
 }
 
+// handleIgnoreFile handles ignoring the file.
+// g and v are passed by the gocui.
+// returns an error if something goes wrong.
 func (gui *Gui) handleIgnoreFile(g *gocui.Gui, v *gocui.View) error {
-	file, err := gui.getSelectedFile(g)
+
+	file, err := gui.getSelectedFile()
 	if err != nil {
-		return gui.createErrorPanel(g, err.Error())
+		err = gui.createErrorPanel(err.Error())
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleIgnoreFile: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
+
 	if file.Tracked {
-		return gui.createErrorPanel(g, gui.Tr.SLocalize("CantIgnoreTrackFiles"))
+
+		err = gui.createErrorPanel(gui.Tr.SLocalize("CantIgnoreTrackFiles"))
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleIgnoreFile: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
 	if err := gui.GitCommand.Ignore(file.Name); err != nil {
-		return gui.createErrorPanel(g, err.Error())
+
+		err = gui.createErrorPanel(err.Error())
+		if err != nil {
+			gui.Log.Errorf("Failed to create error panel at handleIgnoreFile: %s\n", err)
+			return err
+		}
+
+		return nil
 	}
-	return gui.refreshFiles(g)
-}
 
-func (gui *Gui) renderfilesOptions(g *gocui.Gui, file *commands.File) error {
-	return gui.renderGlobalOptions(g)
+	err = gui.refreshFiles()
+	if err != nil {
+		gui.Log.Errorf("Failed to refresh files at handleIgnoreFile: %s\n", err)
+		return err
+	}
+
+	return nil
 }
 
-func (gui *Gui) handleFileSelect(g *gocui.Gui, v *gocui.View) error {
-	file, err := gui.getSelectedFile(g)
+// handleFileSelect is called when a file is selected.
+// It checks if there are any changed files and if there is one
+// and it is selected, it gets rendered into the main view
+func (gui *Gui) handleFileSelect() error {
+
+	var content string
```

Comment:
I'm not sure why this variable was initially declared before being initialized, but I think it may as well just be declared/initialized in the one place i.e. using `:=`

Reply:
Agreed, i'll change that